metrics/sources/summary/summary.go

```diff
@@ -234,6 +241,64 @@ func (this *summaryMetricsSource) decodeContainerStats(podLabels map[string]stri
 	return containerMetrics
 }
 
+// decodePVCStats computes PVC metrics using volume stats of each pod.
+// It selects the volume stats of the pod with the latest scrape time.
+// Metric collection time is the earliest pod start time among all pods.
+// PVC stats are only available when the PVC is in use by at least one pod.
```

Comment:
Did you investigate a possibility of retrieving PVCs not used by any pod? Otherwise, what value do these metrics add? Heapster already collects (and exports) volume metrics on pod level.

Reply:
I think it'd be a lot more difficult to obtain filesystem stats of volumes not mounted on any pod. The main value is to allow users to query PVC stats by PVC name, without having to scan through pods themselves. But maybe adding a `pvc_label` to existing volume metrics is sufficient for this purpose?

Reply:
I think Heapster can retrieve PVCs from api server, using core v1 lister: https://github.com/kubernetes/client-go/blob/master/listers/core/v1/persistentvolumeclaim.go. Similar examples are present in processors, for example pod_based_enricher lists all pods and retrieves some information for them: https://github.com/kubernetes/heapster/blob/master/metrics/processors/pod_based_enricher.go

I prefer this over adding pvc label to existing volume metrics, because this label would be applicable only to particular type of the volume. This would leave us without the association between PVCs and pods, but do we need it?

Reply:
Actually, I missed a part of it (api server don't have most data about the volumes that we need). So this approach LGTM, this comment is resolved.